In [3]:
import pandas as pd

In [4]:
df = pd.read_csv("Dataset .csv")

In [5]:
df.dtypes

Restaurant ID             int64
Restaurant Name          object
Country Code              int64
City                     object
Address                  object
Locality                 object
Locality Verbose         object
Longitude               float64
Latitude                float64
Cuisines                 object
Average Cost for two      int64
Currency                 object
Has Table booking        object
Has Online delivery      object
Is delivering now        object
Switch to order menu     object
Price range               int64
Aggregate rating        float64
Rating color             object
Rating text              object
Votes                     int64
dtype: object

In [9]:
num_col=['Average Cost for two','Price range','Aggregate rating','Votes']

In [10]:
df[num_col].describe()

Average Cost for two  Price range  Aggregate rating         Votes
count           9551.000000  9551.000000       9551.000000   9551.000000
mean            1199.210763     1.804837          2.666370    156.909748
std            16121.183073     0.905609          1.516378    430.169145
min                0.000000     1.000000          0.000000      0.000000
25%              250.000000     1.000000          2.500000      5.000000
50%              400.000000     2.000000          3.200000     31.000000
75%              700.000000     2.000000          3.700000    131.000000
max           800000.000000     4.000000          4.900000  10934.000000

## Insights

- Average Cost for two and votes are having high standard deviation and maximum values
- The median cost (400) and median votes (31) shows that most restaurants are having modest pricing and visibility
- price range mostly falls between 1 and 2 which shows affordability in dataset.
  

In [11]:
df['Country Code'].nunique()

15

In [12]:
df['Country Code'].value_counts()

Country Code
1      8652
216     434
215      80
30       60
189      60
214      60
148      40
208      34
14       24
162      22
94       21
166      20
184      20
191      20
37        4
Name: count, dtype: int64

In [14]:
country_map = {
    1: 'India',
    14: 'Australia',
    30: 'Brazil',
    37: 'Canada',
    94: 'Indonesia',
    148: 'New Zealand',
    162: 'Philippines',
    166: 'Qatar',
    184: 'Singapore',
    189: 'South Africa',
    191: 'Sri Lanka',
    208: 'Turkey',
    214: 'UAE',
    215: 'United Kingdom',
    216: 'United States'
}


In [15]:
df['Country']=df['Country Code'].map(country_map)

In [19]:
df['Country'].value_counts()


Country
India             8652
United States      434
United Kingdom      80
Brazil              60
South Africa        60
UAE                 60
New Zealand         40
Turkey              34
Australia           24
Philippines         22
Indonesia           21
Qatar               20
Singapore           20
Sri Lanka           20
Canada               4
Name: count, dtype: int64

In [20]:
df['City'].nunique()

141

In [22]:
df['City'].value_counts()

City
New Delhi         5473
Gurgaon           1118
Noida             1080
Faridabad          251
Ghaziabad           25
                  ... 
Lakes Entrance       1
Mohali               1
Panchkula            1
Bandung              1
Randburg             1
Name: count, Length: 141, dtype: int64

In [23]:
df['Cuisines'].nunique()

1825

In [24]:
df['Cuisines'].value_counts()

Cuisines
North Indian                       936
North Indian, Chinese              511
Chinese                            354
Fast Food                          354
North Indian, Mughlai              334
                                  ... 
Kebab, Izgara                        1
World Cuisine                        1
World Cuisine, Mexican, Italian      1
Kebab, Turkish Pizza, D�_ner         1
Turkish Pizza                        1
Name: count, Length: 1825, dtype: int64

### Summary: Categorical Variable Exploration

- **Country Code**:
  - 15 unique countries
  - India (Code 1) dominates with 8652 restaurants
  - Added a new `Country` column using a mapping dictionary

- **City**:
  - 141 unique cities
  - New Delhi leads with 5473 restaurants
  - Data is heavily concentrated in Indian metro areas

- **Cuisines**:
  - 1825 unique cuisine combinations
  - North Indian is the most common cuisine (936 entries)
  - Long-tail distribution with many unique combinations used only once
